Akka.Serialization.MessagePack/Formatters/PolymorphicFormatter.cs

```diff
@@ -17,7 +17,11 @@ public PolymorphicFormatter(IMessagePackFormatter<T> normalFormatter)
         public void Serialize(ref MessagePackWriter writer, T value,
             MessagePackSerializerOptions options)
         {
-            if (value.GetType() == typeof(T))
+            if (value == null)
+            {
+                writer.WriteNil();
+            }
+            else if (value.GetType() == typeof(T))
```

Comment:
This helps some optimizations and probably adds some needed safety (I do think Messagepack has it's own null protection but this helped perf since JITter knows `.GetType()` won't throw, and at best fixes a hazard I haven't seen previously.)

Semi related is that valuetype of `default` should be OK here, since `null` is not the same as default in generic context.

There may still be a better way to handle this (opt and more safety), open to suggestions!